src/client/pmix_client_get.c

```diff
@@ -83,14 +83,6 @@ PMIX_EXPORT pmix_status_t PMIx_Get(const pmix_proc_t *proc, const char key[],
     pmix_cb_t *cb;
     pmix_status_t rc;
 
-    PMIX_ACQUIRE_THREAD(&pmix_global_lock);
-
-    if (pmix_globals.init_cntr <= 0) {
-        PMIX_RELEASE_THREAD(&pmix_global_lock);
-        return PMIX_ERR_INIT;
-    }
-    PMIX_RELEASE_THREAD(&pmix_global_lock);
-
```

Comment:
I take it you are then relying on checking for initialization in the non-blocking call? If so, we probably should review the code that executes prior to calling the nb function to ensure that nothing there relies on PMIx being initialized

Reply:
Actually, you don't have to look far - line 95 (in your changed version) calls PMIX_NEW which requires that the object system be initialized. I'm afraid the lock has to remain.

Reply:
This change has been rejected.